File: benchmark.py
import pandas as pd
import plotly.graph_objects as go
import plotly.express as px
from plotly.subplots import make_subplots


df = pd.read_json("data.json", orient='split');

fig = make_subplots(rows=1, cols=1, x_title="Empty Spaces on the board", y_title="Milliseconds (ms)", figure =
    go.Figure(layout = {
        "title": "The time taken for a minimax search against the number of empty spaces to search through",
        "font": dict(
            family="Roboto Mono",
            size=14,
            color="Black"
        )
    })
);


fig.add_trace(go.Scatter(name = "Empty Space", y = df["Empty_Space"], x = df["columns"], fill='none', mode='lines+markers'), row=1, col=1);


fig.write_html('graph.html', auto_open=True)

File: benchmark.js
var RowsNum;
var ColumnsNum;
var rows = [];
var cols = [];
var data = [];


function initJSON(){
    RowsNum = rows.length;
    ColumnsNum = 1;

    cols = ["Empty_Space"];

    // for (var i=0; i<RowsNum; rows[i]=_rows[i], ++i);
    // for (var i=0; i<ColumnsNum; cols[i]=_cols[i], ++i);
    // for (var i=0; i<RowsNum; ++i){
    //     data.push([]);
    //     for (var j=0; j<ColumnsNum; data[i][j]=0.0, ++j);
    // }
}


var outputStream;
function flush(){
    outputStream = "";
    outputStream += "{";

    /// write columns
    outputStream += "\"columns\":[\"columns\",";
    for (var i=0; i<ColumnsNum-1; ++i){
        outputStream += "\"" + cols[i] + "\",";
    }
    outputStream += "\"" + cols[ColumnsNum-1] + "\"" + "],";

    /// write rows
    outputStream += "\"index\":[";
    for (var i=0; i<RowsNum-1; ++i){
        outputStream += "\"" + i + "\",";
    }
    outputStream += "\"" + (RowsNum-1) + "\"" + "],";

    /// write data
    outputStream += "\"data\":[";
    for (var i=0; i<RowsNum; ++i){
        outputStream += "[" + "\"" + rows[i] + "\",";
        for (var j=0; j<ColumnsNum; ++j){
            outputStream += "\"" + data[i][j] + "\"";
            if (j < ColumnsNum-1){ outputStream += ","; };
        }
        outputStream += "]";
        if (i < RowsNum-1){ outputStream += ","; };
    }
    outputStream += "]";
    outputStream += "}";

    console.log(outputStream);
}

var rowIndex = 0;
function insertRow(v){
    rows.push(v.rowVal);
    data.push([v.data]);
    rowIndex++;
}
